webapp/src/projects.tsx

```diff
@@ -489,43 +515,57 @@ class HeroBanner extends data.Component<ISettingsProps, HeroBannerState> {
                 this.prevGalleries = pxt.Util.concat(res.map(g => g.cards))
                     .filter(card => card.url)
                     .slice(0, 5); // max 5 cards
+                if (heroBanner) {
+                    this.prevGalleries.unshift(heroBanner);
+                }
                 this.startRefresh();
             }
         }
-        return this.prevGalleries || [];
+        return this.prevGalleries;
     }
 
     renderCore() {
         const targetTheme = pxt.appTarget.appTheme;
         const { cardIndex } = this.state;
-        const showHeroBanner = !!targetTheme.homeScreenHero;
-        if (!showHeroBanner)
-            return null; // nothing to see here
+        const isGallery = !!targetTheme.homeScreenHeroGallery;
+
         const cards = this.fetchGallery();
-        const card = (cardIndex !== undefined && cards[cardIndex]) || {
-            imageUrl: targetTheme.homeScreenHero
-        }
+        const card = cards[cardIndex];
+        if (!card)
+            return null; // nothing to see here
+
         const handleSetCard = (i: number) => () => this.handleSetCardIndex(i)
         let url = card.url;
         // open tutorials in browser
-        if (card.cardType === "tutorial" && /^\//.test(url))
-            url = `#tutorial:${url}`
+        if (card.cardType === "tutorial" && /^\//.test(url)) {
```

Comment:
are we only handling tutorials for now? will this break if someone puts an example card in the gallery?

Reply:
It'll point it at the url in there -- so we can point at anything from the md (this will point at a 'hero' gallery or something of the sort that won't be a normal gallery).

You're right though that should probably just take the same behavior from the detail view, I'll see how much work it is to pull that out

Reply:
Okay, pulled out the logic for the detail buttons in https://github.com/microsoft/pxt/pull/8171/commits/192698ad9ed1732216c31e493b0f0cc8d942b151 so should work with all cardTypes -- not the prettiest but that path is a bit messy already / probably put off cleaning it up for a later PR?